src/screens/TestCases/Modal/CustomFilterModal.jsx

```diff
@@ -0,0 +1,420 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { Overlay, Tooltip, Modal, Button } from 'react-bootstrap';
+import './CustomFilterStyle.css'; // Ensure this CSS file includes the .custom-container class
+import { getDraftTestCaseList } from '../../../redux/services/testCases/downloadFormatFile';
+import { useDispatch } from 'react-redux';
+
+const CustomFilterModal = ({
+  show,
+  handleClose,
+  handleApply,
+  position,
+  filterColumn,
+  handleCheckboxChange,
+  handleBetweenValueChange,
+  selectedFilters,
+  handleSelectAll,
+  uniqueValues,
+  searchTerm,
+  setSearchTerm,
+  paginationData,
+  filterColumnId,
+  setFilterType,
+  setFilterText,
+  filterType,
+  columnName,
+  handleAscendingClick,
+  handleDescendingClick,
+  type
+}) => {
+  const [showTooltip, setShowTooltip] = useState(false);
+  const [showModal, setShowModal] = useState(false);
+  const target = useRef(null);
+  const searchRef = useRef(null);
+  const containerRef = useRef(null);
+  // const [filterType, setFilterType] = useState('');
+  // const [filterText, setFilterText] = useState('');
+
+  const textFilterData = [
+    {
+      value: 'equals',
+      label: 'Equals'
+    },
+    {
+      value: 'does not equal',
+      label: 'Does Not Equals'
+    },
+    {
+      value: 'begins with',
+      label: 'Begins With'
+    },
+    {
+      value: 'does not begin with',
+      label: 'Does Not Begin With'
+    },
+    {
+      value: 'ends with',
+      label: 'End with'
+    },
+    {
+      value: 'does not begin with',
+      label: 'Does Not End with'
+    },
+    {
+      value: 'contains',
+      label: 'Contains'
+    },
+    {
+      value: 'does not contain',
+      label: 'Does Not Contain'
+    }
+  ];
+
+  const numberFilterData = [
+    {
+      value: 'equals',
+      label: 'Equals'
+    },
+    {
+      value: 'is greater than',
+      label: 'Is Greater Than'
+    },
+    {
+      value: 'is between',
+      label: 'Between'
+    },
+    {
+      value: 'is not between',
+      label: 'Is Not Between'
+    }
+  ];
+
+  // <option value="equals">Equals</option>
+  // <option value="does not equal">
+  //   Does Not Equals
+  // </option>
+  // <option value="begins with">Begins With</option>
+  // <option value="does not begin with">Does Not Begin With</option>
+
+  // <option value="ends with">End with</option>
+  // <option value="does not end with">Does Not End with</option>
+
+  // <option value="contains"> Contains</option>
+  // <option value="does not contain">
+  //   Does Not Contain
+  // </option>
+
+  useEffect(() => {
```

Comment:
Mention Purpose of this code

Reply:
to show the filter modal inside the filter icon